cmake/helpers/CheckDependentLibraries.cmake

```diff
@@ -449,6 +449,23 @@ gdal_check_package(MONGOCXX "Enable MongoDBV3 driver" CAN_DISABLE)
 define_find_package2(HEIF libheif/heif.h heif PKGCONFIG_NAME libheif)
 gdal_check_package(HEIF "HEIF >= 1.1" CAN_DISABLE)
 
+include(CheckCXXSourceCompiles)
+check_cxx_source_compiles(
+    "
+    #include <libheif/heif.h>
+    int main()
+    {
+        struct heif_image_tiling tiling;
+        return 0;
+    }
+    "
+    LIBHEIF_SUPPORTS_TILES
+)
+if (LIBHEIF_SUPPORTS_TILES)
+    set(LIBHEIF_SUPPORTS_TILES 1)
+    add_definitions(-DLIBHEIF_SUPPORTS_TILES)
```

Comment:
it would be if we could avoid a general directive like ``add_definitions()`` that will apply to the whole GDAL code base and use instead something like:
```
    set_property(TARGET HEIF::HEIF APPEND PROPERTY
                 INTERFACE_COMPILE_DEFINITIONS "LIBHEIF_SUPPORTS_TILES")
```

Reply:
Should be fixed.